lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -998,38 +1017,70 @@ namespace Js
     bool WebAssemblyArrayBuffer::IsValidVirtualBufferLength(uint length) const
     {
 #if ENABLE_FAST_ARRAYBUFFER
-        return true;
+        return CONFIG_FLAG(WasmFastArray);
 #else
         return false;
 #endif
     }
 
-    ArrayBuffer * WebAssemblyArrayBuffer::TransferInternal(uint32 newBufferLength)
+    WebAssemblyArrayBuffer* WebAssemblyArrayBuffer::GrowMemory(uint32 newBufferLength)
```

Comment:
While we're at it, it's probably worth it to make this non-throwing. In case of failure return nullptr, and if we get nullptr then return -1.